env0/provider.go

```diff
@@ -13,6 +13,14 @@ import (
 
 func Provider(version string) plugin.ProviderFunc {
 	return func() *schema.Provider {
+		apiKeyEnv := "ENV0_API_KEY"
```

Comment:
This is required when running the provider unit test (which modified the env).
Unfortunately, there's is no good way to configure env variables when running tests in parallel.
E.g. the new t.SetEnv(...) method is not supported when running tests in parallel (maybe future golang versions will have better support for it).

Reply:
It'd be helpful if you can add some comment around that so any other person that reaches this piece of logic will understand what's going on